src/providers/ses.rs

```diff
@@ -72,14 +72,18 @@ impl Provider for SesProvider {
         request.message = message;
         request.source = self.sender.to_string();
 
-        match self.client.send_email(&request).sync() {
-            Ok(response) => Ok(response.message_id),
-            Err(error) => {
-                println!("SES error: {}", error);
-                Err(ProviderError {
-                    description: error.to_string(),
-                })
-            }
+        self.client
+            .send_email(&request)
+            .sync()
+            .map(|response| response.message_id)
+            .map_err(From::from)
```

Comment:
A visible sign of progress here. I didn't know about the `From` trait when I first wrote this code, so took the opportunity to tidy it up while I was passing through.